### DISCUSSION TITLE
console application with quartz.net sheduler

### DISCUSSION THREAD

**Discussion by sruthi-kurup** [quartznet/quartznet#1323] [Q&A]
hi @lahma 

I'm working on an asp application. For that I have created jobs and scheduled jobs and used pgsql persistent store. Used ado configs to store jobs in persistant store.

Configured a custom jobfactory also

also I used prism's IEventAggregator in IJob class to do the task. But unfortunately the IEventAggregator is not subscribing the job. While debugging I understood that this is due to the issue of the service provider configuration in the  custom job factory.

So now im trying to create a console application with all the quartz scheduling features and pgsql persistance job. 

I created the console app with the simple scheduler and default job factory, without any persistance store.

I have seen so many tutorials about using quartz as windows servce. But i dont want to use it as a service. 

Is there any way to do the console app with the quartz scheduler features.

Thankyou

**Comment by lahma** [comment]
There's [example console project](https://github.com/quartznet/quartznet/tree/main/src/Quartz.Examples.Worker), you can also check [the ASP.NET Core sample](https://github.com/quartznet/quartznet/tree/main/src/Quartz.Examples.AspNetCore) for more configuration options.

**Reply by sruthi-kurup** [reply]
thanks for the reply

**Reply by sruthi-kurup** [reply]
@lahma i had referred the code, but how can we use the npg persistent store here and also schedule job on user request. Not to schedule job on application start
